debezium-core/src/main/java/io/debezium/pipeline/metrics/PipelineMetrics.java

```diff
@@ -49,26 +53,43 @@
     }
 
     @Override
-    public void onEvent(DataCollectionId source, OffsetContext offset, Object key, Struct value) {
-        updateCommonEventMetrics();
+    public void onEvent(DataCollectionId source, OffsetContext offset, Object key, Struct value, Operation operation) {
+        updateCommonEventMetrics(operation);
         lastEvent = metadataProvider.toSummaryString(source, offset, key, value);
     }
 
-    private void updateCommonEventMetrics() {
+    private void updateCommonEventMetrics(Operation operation) {
         totalNumberOfEventsSeen.incrementAndGet();
         lastEventTimestamp.set(clock.currentTimeInMillis());
+        if (operation != null) {
+            switch (operation) {
+                case READ:
+                case CREATE:
```

Comment:
That feels not quite right; I'd say we should either separate read (i.e. snapshot) and create events via two different metrics, or only count create events (as the number of snapshot events is already present via [rows scanned](https://debezium.io/documentation/reference/stable/connectors/mysql.html#connectors-snaps-metric-rowsscanned_mysql) in snapshot metrics.